model/src/wav_comp_nuopc.F90

```diff
@@ -0,0 +1,1416 @@
+module wav_comp_nuopc
+
+  !/ ------------------------------------------------------------------- /
+  !/
+  !/                  +-----------------------------------+
+  !/                  | WAVEWATCH III           NOAA/NCEP |
+  !/                  |           H. L. Tolman            |
```

Comment:
It seems that the code documentation is being updated to utilize doxygen? If so, I think it makes more sense to move directly to inserting the required doxygen formatting.

Reply:
I'm familiar w/ doxygen because I needed to add it for the grid generation code being added to ufs-utils. Does the documentation get built automatically for every push, as w/ ufs-utils?

Reply:
@DeniseWorthen We have not hooked in the build automation yet.  I'm actually working right now on a getting a current draft PR with the first batch of marked-up files ready for submission.  This will happen later today or tomorrow.  After that we'll be in a position to set up the auto build.

Reply:
I've started adding the doxygen documentation to a separate branch which I will merge in to this PR branch when I've got the majority of it done. I can build it by hijacking the documentation build in my ufs-utils fork. 

To start, I used the ``ww3_doxy_tmpl.md`` file for the required features. What I found though is that if add both ``@param`` and then also ``!< `` to document the variables, it ends up "double documenting". For example, this code snip:

```
!> Determine if a field is in a state
!!
!! @param[in] State    an ESMF_State
!! @param[in] fldname  the name of the ESMF field
!!
!> @author mvertens@ucar.edu, Denise.Worthen@noaa.gov  @<01-05-2022>
  logical function state_fldchk(State, fldname)
    ! ----------------------------------------------
    ! Determine if field is in state
    ! ----------------------------------------------

    ! input/output variables
    type(ESMF_State) , intent(in)  :: State                  !< ESMF_State object
    character(len=*) , intent(in)  :: fldname                !< a field name
```

produces the following
<img width="780" alt="Screen Shot 2022-04-05 at 8 08 10 AM" src="https://user-images.githubusercontent.com/40498404/161751123-422bd94e-b4c5-49eb-b7d3-d29cc07dfe6d.png">

I wonder if it makes more sense to use @param only (or vice-versa) since documenting the arguments twice is redundant.

Reply:
kudos for adding the doxygen markup!

Reply:
@MatthewMasarik-NOAA Regarding module variables, do variables which are local only need to be docmented? That would mean things like ``integer :: i,j`` ?

Reply:
@DeniseWorthen , variables local to subroutines and functions don't need to be documented. Just variables that are listed between the 'module' and 'contains' keywords.  Does that answer it?

Reply:
When I said 'all' above, that's a little misleading ;)  only those non-locals.